opal/mca/btl/smcuda/btl_smcuda_accelerator.c

```diff
@@ -79,6 +83,8 @@ int mca_btl_smcuda_accelerator_init(void)
         goto cleanup_and_error;
     }
 
+    opal_finalize_register_cleanup(mca_btl_smcuda_accelerator_fini);
```

Comment:
I believe you when you say this will fix the issue, but do you mind explaining why it needs to happen this way since I don't quite understand. The pml ob1 component does something similar and I'm wondering if it requires the same fix

Reply:
@wckzhang does this comment block help answer your questions?

Reply:
Yes, I think we need to do the same thing with ob1 then

Reply:
isn't in theory this the case for all components that use the accelerator framework functionality?

Reply:
@wckzhang where would you want to make a change in the ob1 to use the opal cleanup callback mechanism?

Reply:
@edgargabriel if you find your consumer of the accelerator framework is doing something in its component close method that involves calls into the accelerator framework then you are probably right.

Reply:
I take it back about ob1 because it already does it in a fini function instead of component close

Reply:
@edgargabriel Yes, though the other components aren't doing any accelerator functionality during finalization. The reason ob1 and smcuda are susceptible to this is because they create event arrays for the purpose of their asynchronous progress engine which need to be cleaned up during finalization.

Reply:
what about the rcache/grdma component? I *think* that one does something similar, isn't it?

Reply:
No as far as I can tell it doesn't access the accelerator framework in its closing sequence, just during its register path.